### ISSUE TITLE
[Unstable V0.1100.0.2] Crash on EnemyAIController.UpdateEscape

### ISSUE THREAD

**Issue by Rokvach** [Regalis11/Barotrauma#4217]
From Discord

No clear knowledge of the cause nor reproduction steps

> JP
> Stopped at a wreck (Kastrull) on the way to the Outpost and the game crashed when I was leaving the wreck...
> frames went down to 14 fps (and probably lower) then game crashed

```
Barotrauma Client crash report (generated on 04/11/2020 7:27:30 PM)


Barotrauma seems to have crashed. Sorry for the inconvenience! 


4718BDBDB44333D92EEC866BF817DC67


Game version 0.1100.0.2 (UnstableWindows, branch dev, revision 95c793eef6)
Graphics mode: 1920x1080 (BorderlessWindowed)
VSync ON
Language: English
Selected content packages: Vanilla 0.9, Malaco_1.068, Carrack, Defiant D_1.45
Level seed: Harmonia LineaSkadi's Cauldron
Loaded submarine: Malaco_1.095_cu (F6CB24A90A27AD1D6FF4B7A4B046F206)
Selected screen: Barotrauma.GameScreen
SteamManager initialized


System info:
    Operating system: Microsoft Windows NT 10.0.18363.0 64 bit
    GPU name: Intel(R) UHD Graphics 630
    Display mode: {Width:1920 Height:1080 Format:Color AspectRatio:1.7777778}
    GPU status: Normal


Exception: Object reference not set to an instance of an object. (System.NullReferenceException)
Target site: Void UpdateEscape(Single)
Stack trace: 
   at Barotrauma.EnemyAIController.UpdateEscape(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\EnemyAIController.cs:line 688
   at Barotrauma.EnemyAIController.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\EnemyAIController.cs:line 447
   at Barotrauma.AICharacter.Update(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AICharacter.cs:line 90
   at Barotrauma.Character.UpdateAll(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 2319
   at Barotrauma.GameScreen.Update(Double deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 175
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 927
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 656
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 92
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Microsoft.Xna.Framework.Game.Run() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 367
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58


Last debug messages:
[04/11/2020 6:16:37 PM] Generating an outpost for the start of the level... (Location: Harmonia Linea, level type: LocationConnection)
[04/11/2020 6:16:37 PM] Generating an outpost for the end of the level... (Location: Skadi's Cauldron, level type: LocationConnection)
[04/11/2020 6:03:06 PM] Generating an outpost for the start of the level... (Location: Harmonia Linea, level type: LocationConnection)
[04/11/2020 6:03:05 PM] Generating an outpost for the end of the level... (Location: Skadi's Cauldron, level type: LocationConnection)
[04/11/2020 5:48:30 PM] Generating an outpost for the start of the level... (Location: Harmonia Linea, level type: LocationConnection)
[04/11/2020 5:48:29 PM] Generating an outpost for the end of the level... (Location: Skadi's Cauldron, level type: LocationConnection)
[04/11/2020 5:24:00 PM] Generating an outpost for the start of the level... (Location: Harmonia Linea, level type: LocationConnection)
[04/11/2020 5:23:58 PM] Generating an outpost for the end of the level... (Location: Skadi's Cauldron, level type: LocationConnection)
[04/11/2020 5:23:46 PM] (Msg) RelayNetWorkStatus:  avail=OK  config=OK  anyrelay=OK   (OK)
[04/11/2020 5:23:46 PM] (Msg) Ping location: iad=91+9,atl=192+19/104+9,ord=165+16/110+9,okc=164+16/130+9,sea=221+22/152+9,lax=185+18/155+9,eat=199+19/155+9,par=231+23/168+9,lim=171+17,fra=250+25/177+9,gru=268+26/235+9,sgp=324+32/334+9
[04/11/2020 5:23:46 PM] (Msg) Ping measurement completed
[04/11/2020 5:23:46 PM] (Warning) SteamNetworkingSockets lock held for 6.8ms.  (Performance warning).  ServiceThread,SendUDPacket
[04/11/2020 5:23:45 PM] (Debug) Destroying relay 'sof#2 (139.45.193.10:27016)' because initial_ping_timeout
[04/11/2020 5:23:43 PM] (Warning) SteamNetworkingSockets lock held for 5.5ms.  (Performance warning).  ServiceThread,SendUDPacket
[04/11/2020 5:23:42 PM] (Debug) Destroying relay 'sof#5 (139.45.193.10:27019)' because initial_ping_timeout
[04/11/2020 5:23:41 PM] (Warning) SteamNetworkingSockets lock held for 8.6ms.  (Performance warning).  ServiceThread,SendUDPacket
[04/11/2020 5:23:41 PM] Loading additional level object prefabs from file 'Content/Map/Biomes/TheGreatSea/TheGreatSeaObjects.xml'
[04/11/2020 5:23:41 PM] Loading additional level object prefabs from file 'Content/Map/Biomes/HydrothermalWastes/HydrothermalWastesObjects.xml'
[04/11/2020 5:23:41 PM] Loading additional level object prefabs from file 'Content/Map/Biomes/EuropanRidge/EuropanRidgeObjects.xml'
[04/11/2020 5:23:41 PM] Loading additional level object prefabs from file 'Content/Map/Biomes/ColdCaverns/ColdCavernsObjects.xml'
[04/11/2020 5:23:41 PM] Loading additional level object prefabs from file 'Content/Map/Biomes/AphoticPlateau/AphoticPlateauObjects.xml'
[04/11/2020 5:23:41 PM] Loading additional level object prefabs from file 'Content/Map/Biomes/IceCave/IceCaveObjects.xml'
[04/11/2020 5:23:39 PM] (Debug) Destroying relay 'sof#3 (139.45.193.10:27017)' because initial_ping_timeout
[04/11/2020 5:23:39 PM] (Warning) SteamNetworkingSockets lock held for 9.8ms.  (Performance warning).  ServiceThread,SendUDPacket(x2)
[04/11/2020 5:23:38 PM] (Warning) SteamNetworkingSockets lock held for 14.2ms.  (Performance warning).  ServiceThread,SendUDPacket(x2)
[04/11/2020 5:23:38 PM] (Warning) SteamNetworkingSockets lock held for 20.8ms.  (Performance warning).  ServiceThread,RecvUDPPacket,SendUDPacket(x3)
[04/11/2020 5:23:38 PM] (Warning) SteamNetworkingSockets lock held for 59.5ms.  (Performance warning).  ServiceThread,RecvUDPPacket,SendUDPacket(x4)
[04/11/2020 5:23:38 PM] (Warning) SteamNetworkingSockets lock held for 18.2ms.  (Performance warning).  ServiceThread,SendUDPacket(x5)
[04/11/2020 5:23:38 PM] (Warning) SteamNetworkingSockets lock held for 13.9ms.  (Performance warning).  ServiceThread,SendUDPacket(x2)
[04/11/2020 5:23:38 PM] (Warning) SteamNetworkingSockets lock held for 48.1ms.  (Performance warning).  ServiceThread,SendUDPacket(x5)
[04/11/2020 5:23:38 PM] (Warning) SteamNetworkingSockets lock held for 92.7ms.  (Performance warning).  ServiceThread,RecvUDPPacket,SendUDPacket(x8)
[04/11/2020 5:23:38 PM] (Warning) SteamNetworkingSockets lock held for 108.6ms.  (Performance warning).  ServiceThread,SendUDPacket(x9)
[04/11/2020 5:23:38 PM] (Warning) SteamNetworkingSockets lock held for 175.7ms.  (Performance warning).  ServiceThread,SendUDPacket(x13)
[04/11/2020 5:23:37 PM] (Warning) SteamNetworkingSockets lock held for 139.3ms.  (Performance warning).  ServiceThread,RecvUDPPacket,SendUDPacket(x12)
[04/11/2020 5:23:37 PM] (Warning) SteamNetworkingSockets lock held for 191.0ms.  (Performance warning).  ServiceThread,SendUDPacket(x18)
[04/11/2020 5:23:37 PM] (Warning) SteamNetworkingSockets lock held for 160.6ms.  (Performance warning).  ServiceThread,RecvUDPPacket,SendUDPacket(x14),IRawUDPSocket::Close,OpenRawUDPSocketInternal
[04/11/2020 5:23:37 PM] (Debug) Destroying relay 'sof#1 (139.45.193.10:27015)' because initial_ping_timeout
[04/11/2020 5:23:37 PM] (Verbose) Communications with routing cluster 'iad' established.
[04/11/2020 5:23:37 PM] (Warning) SteamNetworkingSockets lock held for 51.1ms.  (Performance warning).  ServiceThread,SendUDPacket(x9)
[04/11/2020 5:23:36 PM] (Warning) SteamNetworkingSockets lock held for 53.5ms.  (Performance warning).  ServiceThread,RecvUDPPacket(x11),SendUDPacket(x10)
[04/11/2020 5:23:36 PM] (Warning) SteamNetworkingSockets lock held for 216.3ms.  (Performance warning).  ServiceThread,SendUDPacket(x20),IRawUDPSocket::Close
[04/11/2020 5:23:36 PM] (Debug) Destroying relay 'iad#223 (162.254.192.86:27031)' because initial_ping_timeout
[04/11/2020 5:23:36 PM] (Warning) SteamNetworkingSockets lock held for 157.5ms.  (Performance warning).  ServiceThread,SendUDPacket(x11)
[04/11/2020 5:23:36 PM] (Warning) SteamNetworkingSockets lock held for 96.2ms.  (Performance warning).  ServiceThread,SendUDPacket(x10)
[04/11/2020 5:23:36 PM] (Warning) SteamNetworkingSockets lock held for 69.6ms.  (Performance warning).  ServiceThread,RecvUDPPacket,SendUDPacket(x9)
[04/11/2020 5:23:36 PM] (Warning) SteamNetworkingSockets lock held for 58.6ms.  (Performance warning).  ServiceThread,SendUDPacket(x10)
[04/11/2020 5:23:36 PM] (Warning) SteamNetworkingSockets lock held for 85.7ms.  (Performance warning).  ServiceThread,SendUDPacket(x10)
[04/11/2020 5:23:36 PM] (Warning) SteamNetworkingSockets lock held for 204.5ms.  (Performance warning).  ServiceThread,SendUDPacket(x16)
[04/11/2020 5:23:35 PM] (Warning) SteamNetworkingSockets lock held for 130.3ms.  (Performance warning).  ServiceThread,SendUDPacket(x13)
[04/11/2020 5:23:35 PM] (Warning) SteamNetworkingSockets lock held for 178.3ms.  (Performance warning).  ServiceThread,SendUDPacket(x17)
[04/11/2020 5:23:35 PM] (Verbose) Communications with routing cluster 'waw' established.
[04/11/2020 5:23:35 PM] (Verbose) Communications with routing cluster 'tyo1' established.
[04/11/2020 5:23:35 PM] (Verbose) Communications with routing cluster 'tyo' established.
[04/11/2020 5:23:35 PM] (Verbose) Communications with routing cluster 'syd' established.
[04/11/2020 5:23:35 PM] (Warning) SteamNetworkingSockets lock held for 147.8ms.  (Performance warning).  ServiceThread,RecvUDPPacket,SendUDPacket(x14)
[04/11/2020 5:23:35 PM] (Verbose) Communications with routing cluster 'vie' established.
[04/11/2020 5:23:35 PM] (Verbose) Communications with routing cluster 'sto2' established.
[04/11/2020 5:23:35 PM] (Verbose) Communications with routing cluster 'sgp' established.
[04/11/2020 5:23:35 PM] (Warning) SteamNetworkingSockets lock held for 80.4ms.  (Performance warning).  ServiceThread,SendUDPacket(x10)
[04/11/2020 5:23:35 PM] (Verbose) Communications with routing cluster 'sto' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'sea' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'scl' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'par' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'ord' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'okc' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'mad' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'maa' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'lux' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'lim' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'lhr' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'lax' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'jnb' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'hkg' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'gru' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'fra' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'eat' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'dxb' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'bom' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'atl' established.
[04/11/2020 5:23:34 PM] (Verbose) Communications with routing cluster 'ams' established.
[04/11/2020 5:23:34 PM] (Warning) SteamNetworkingSockets lock held for 743.6ms.  (Performance warning).  ServiceThread,OpenRawUDPSocketInternal(x30),SendUDPacket(x30)
[04/11/2020 5:23:34 PM] (Msg) RelayNetWorkStatus:  avail=Attempting  config=OK  anyrelay=Attempting   (Performing ping measurement)
[04/11/2020 5:23:34 PM] (Msg) Performing ping measurement
[04/11/2020 5:23:34 PM] (Msg) Got SDR network config.  Loaded revision 283 OK
[04/11/2020 5:23:33 PM] (Warning) SDR network config fetch first attempt failed.  HTTP 504.  .  Trying again.
[04/11/2020 5:23:20 PM] Attempting to open ALC device "OpenAL Soft on Speakers (Realtek(R) Audio)"
[04/11/2020 5:23:18 PM] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
[04/11/2020 5:23:18 PM] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
```

**Comment by itchyOwl** [comment]
Fixed in Regalis11/Barotrauma-development@1b6f5f2.

**Comment by Rokvach** [comment]
There hasn't been any sightings of this since it was introduced to Unstable, closing for now.
